### ISSUE TITLE
Suppress private visibility arguments in generate-manual output

### ISSUE THREAD

**Issue by monoxygen** [apple/swift-argument-parser#665]
Arguments that specify `private` visibility, such as:
```swift
@Flag(help: ArgumentHelp(visibility: .private))
var hiddenFromHelp: Bool = false
```

show up in the man page generated by the `generate-manual` command:
```man
.It Fl -hidden-from-help
```

If `private` visibility indicates that help should never be shown for that argument, it seems reasonable and desirable that it also not be visible in man pages.

**ArgumentParser version:** `1.5.0`
**Swift version:** `swift-driver version: 1.110 Apple Swift version 6.0 (swiftlang-6.0.0.4.52 clang-1600.0.21.1.3)`

### Checklist
- [x] If possible, I've reproduced the issue using the `main` branch of this package
- [x] I've searched for [existing GitHub issues](https://github.com/apple/swift-argument-parser/issues)

### Steps to Reproduce
1. Define an argument with `private` visibility, such as the `hiddenFromHelp` argument above
2. Build the client executable at path `<executable>`
3. Run `swift run generate-manual <executable> --output-directory <outputdir>`

### Expected behavior
`hiddenFromHelp` should not be included in the man page generated at `<outputdir>`

### Actual behavior
`hiddenFromHelp` is included in the man page generated at `<outputdir>`

**Comment by heckj** [comment]
Saw the bug listing while I was digging around the repo related to dumping the help output. The arguments are annotated with visibility information up front, but it wasn't evaluated when generating the manual output, so my PR (#667) checks that additional property before generating relevant content. The existing Math example in the test output included "hidden" content, so I've updated the tests to match - the arguments marked as hidden shouldn't be in the resulting output.

I think I've matched style, etc - but more than happy to tweak or change if something is amiss or isn't following a preferred pattern for this additional check.
